packages/@glimmer/manager/lib/internal/defaults.ts

```diff
@@ -0,0 +1,44 @@
+import { buildCapabilities } from '../util/capabilities';
+
+import type { CapturedArguments as Arguments, HelperCapabilities } from '@glimmer/interfaces';
+
+type FnArgs<Args extends Arguments = Arguments> =
+  | [...Args['positional'], Args['named']]
+  | [...Args['positional'], {}];
+
+interface FunctionHelperState<Args extends Arguments = Arguments> {
+  fn: <Return>(...args: FnArgs<Args>) => Return;
+  args: Args;
+}
+
+export class FunctionHelperManager<State extends FunctionHelperState> {
+  capabilities = buildCapabilities({
+    hasValue: true,
+    hasDestroyable: false,
+    hasScheduledEffect: false,
+  }) as HelperCapabilities;
+
+  createHelper(fn: State['fn'], args: State['args']) {
+    return { fn, args };
+  }
+
+  getValue({ fn, args }: State) {
+    let namedKeys = Object.keys(args.named).length;
+
+    if (args.positional.length === 0 && namedKeys === 0) {
+      return fn();
+    }
+
+    let argsForFn: FnArgs<Arguments> = [...args.positional, namedKeys > 0 ? args.named : {}];
```

Comment:
I think this doesn't work when you have positional but not named. You'll still be passing the empty object in that case. I think you need something like

```js
let args = [...args.positional];

if (Object.keys(args.named).length > 0) {
  args.push(args.named);
}

return fn(...args);
```

Reply:
nay,
![image](https://user-images.githubusercontent.com/199018/149247168-fd55bf3b-739d-43bc-b447-637bb1396219.png)

default arguments are still allowed with the tailing `{}` arg

Reply:
This is what I am saying won't work: `function greeting(name, message = "hello")`

Reply:
ahhhh yeah, I gotchya